File: fixtures/scale/kotlin/small/src/textprinter/render/BorderPainter.kt
package textprinter.render

import textprinter.border.Frame
import textprinter.style.Theme

class BorderPainter(private val frame: Frame) {

    fun paint(body: List<String>, theme: Theme, width: Int): String {
        val sb = StringBuilder()
        sb.append(frame.rule(width)).append('\n')
        for (line in body) {
            sb.append(frame.vertical).append(' ')
            sb.append(theme.apply(line))
            sb.append(' ').append(frame.vertical).append('\n')
        }
        sb.append(frame.rule(width))
        return sb.toString()
    }
}

File: fixtures/scale/dart/small/lib/render/border_painter.dart
import '../border/frame.dart';
import '../style/theme.dart';
import 'render_target.dart';

class BorderPainter implements RenderTarget {
  final Frame frame;

  BorderPainter(this.frame);

  @override
  String paint(List<String> body, Theme theme, int width) {
    final sb = StringBuffer();
    sb.writeln(frame.rule(width));
    for (final line in body) {
      sb.write(frame.vertical);
      sb.write(' ');
      sb.write(theme.apply(line));
      sb.write(' ');
      sb.writeln(frame.vertical);
    }
    sb.write(frame.rule(width));
    return sb.toString();
  }
}
